### PR TITLE
add CRIC unittest

### LINKED ISSUE #8794: Create unittest for CRIC service
related to #8786

### PR DESCRIPTION
**Pull request by ticoann**
fixes #8794

### CODE REVIEW

**Comment by ticoann** [comment]
Alan, I want to discuss a couple of things before create the mock unittest.

1. I refactor the CRIC service code to make it easier to use mock. With the change only 2 methods (_CRICUserQuery, _CRICSiteQuery )need to be mocked instead all of them.
2. It seems it is giving different result between SiteDB and CRIC for PNNs to PSNs. (commented in line)
3. It seems there is no user information in CRIC now.

File: test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        _testPNNstoPSNs_
+
+        Test converting PhEDEx Node Names to Processing Site Names
+        """
+        result = self.myCRIC.PNNstoPSNs(['T1_US_FNAL_Disk', 'T1_US_FNAL_Buffer', 'T1_US_FNAL_MSS'])
+        self.assertTrue(result == ['T1_US_FNAL'])
+        result = self.myCRIC.PNNstoPSNs(['T2_UK_London_IC', 'T2_US_Purdue'])
+        print(result)
+        self.assertItemsEqual(result, [u'T3_UK_London_QMUL', u'T2_US_Purdue', u'T3_UK_SGrid_Oxford',
+                                       u'T2_UK_London_IC', u'T3_UK_ScotGrid_GLA', u'T3_UK_London_RHUL'])
```
Comment (ticoann) [inline test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py]: Alan, 
SiteDB returns only ['T2_UK_London_IC', 'T2_US_Purdue'], But CRIC returns [u'T3_UK_London_QMUL', u'T2_US_Purdue', u'T3_UK_SGrid_Oxford', u'T2_UK_London_IC', u'T3_UK_ScotGrid_GLA', u'T3_UK_London_RHUL']

Do you know why?

File: test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    def setUp(self):
+        """
+        Setup for unit tests
+        """
+        super(CRICTest, self).setUp()
+        self.myCRIC = CRIC()
+
+    def testWhoAmI(self):
+        print("XXXXX")
+        print(self.myCRIC.whoAmI())
```
Comment (ticoann) [inline test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py]: This returns empty list, when does it going to be populated in CRIC

File: src/python/WMCore/Services/CRIC/CRIC.py [hunk 3]
```diff
@@ -61,6 +61,27 @@ def _getResult(self, uri, callname="", args=None, unflatJson=True):
...
+
+    def _CRICSiteQuery(self, callname):
+        """
+        :param callname: name of the call
+        :return: dict of the result
+        """
+
+        uri = "/api/cms/site/query/"
+        extraArgs = {"rcsite_state": "ANY"}
+        sitenames = self._getResult(uri, callname=callname, args=extraArgs)
```
Comment (amaltaro) [inline src/python/WMCore/Services/CRIC/CRIC.py]: how about call it siteinfo?

File: test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py (hunk 1 above)
Comment (amaltaro) [reply test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py]: Are you sure? I just checked the sitedb api and I see those 2 PNNs mapping to a bunch of PSNs (apparently to the same as in CRIC)

**Review by amaltaro — APPROVED** [review]
Thank you very much, Seangchan.
CRIC code got much cleaner with your changes. I have only one suggestion for the code, otherwise all the rest looks fine.

File: test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py (hunk 2 above)
Comment (amaltaro) [reply test/python/WMCore_t/Services_t/CRIC_t/CRIC_t.py]: It has been populated already. From what I discussed with Stefano, it seems it does not work when we add a proxy to the request header. When we make the request with the service certificate (different pem for cert and key), that works just fine(?)

File: src/python/WMCore/Services/CRIC/CRIC.py [hunk 4]
```diff
@@ -61,6 +61,27 @@ def _getResult(self, uri, callname="", args=None, unflatJson=True):
             results = unflattenJSON(results)
         return results
 
+    def _CRICUserQuery(self, callname, unflatJson=True):
+        """
+        :param callname: name of the call
```
Comment (amaltaro) [inline src/python/WMCore/Services/CRIC/CRIC.py]: and maybe we say `:param callname: name of the api to be called`?

**Comment by amaltaro** [comment]
test this please

**Comment by amaltaro** [comment]
Seangchan, since I had to made further changes to the CRIC module, I decided to steal these changes and merge them altogether in my PR #8848 
Thanks for taking care of the hard part ;-) Closing this one

**Pull request closed without merging**